### PR TITLE
Add Equal method

### PR DESCRIPTION
**Pull request by cesartw**

### CODE REVIEW

File: validate.go [hunk 1]
```diff
@@ -650,3 +650,45 @@ func (v *Validator) Range(key string, value, min, max int64, message ...string)
...
+
+		if len(vmsgs) != len(omsgs) {
+			return false
+		}
+
+	NEXT:
+		for _, vmsg := range vmsgs {
+			for _, omsg := range omsgs {
+				if vmsg == omsg {
+					continue NEXT
```
Comment (PanagiotisPtr) [inline validate.go:686]: Personally I think a flag + break would be nicer here 
image

File: validate.go [hunk 2]
```diff
@@ -650,3 +650,45 @@ func (v *Validator) Range(key string, value, min, max int64, message ...string)
 		}
 	}
 }
+
+// Equal returns whether two validators are equal.
+func (v *Validator) Equal(o *Validator) bool {
```
Comment (ready4god2513) [inline validate.go:655]: Is there a use case outside of `reflect.DeepEqual`?

File: validate.go (hunk 1 above)
Comment (cesartw) [reply validate.go:686]: We're all entitled to our terrible opinions

**Comment by arp242** [comment]
Perhaps a bit slower (not even sure about that, but that probably doesn't matter anyway) to save ~20 lines of code:

    return v.String() == o.String()

Would have to change String() to sort the errors too (now it just sorts the keys), which probably isn't a bad change anyway.

**Comment by cesartw** [comment]
> return v.String() == o.String()


I don't know about that. It's like comparing errors by doing `erra.Error() == errb.Error()`, it never feels right.
In fact, *Validator is an `error` so it's literally like doing that

Consistency in `String()` would be nice, tho

**Comment by ripexz** [comment]
Is the goal to check if its the same validator, or to check if two validators have the same errors?
Cause if it's the former, you could add a new unexported struct field like `id`, set it to random val in `New()` and then just compare those 2 internal vals in your func 🤔

**Comment by cesartw** [comment]
> to check if two validators have the same errors

mostly for testing

**Review by ready4god2513 — APPROVED** [review]

**Pull request merged**